tests/func/test_import_url.py

```diff
@@ -206,4 +209,45 @@ def test_import_url_dir(tmp_dir, dvc, workspace, stage_md5, dir_md5):
         "  path: dir\n"
     )
 
+
+def assert_local_dir(dvc, tmp_dir, dir_name):
+    assert set(os.listdir(tmp_dir / dir_name)) == {"file", "subdir"}
+    assert (tmp_dir / dir_name / "file").read_text() == "file"
+    assert list(os.listdir(tmp_dir / dir_name / "subdir")) == ["subfile"]
+    assert (tmp_dir / dir_name / "subdir" / "subfile").read_text() == "subfile"
+
     assert dvc.status() == {}
+
+
+def test_import_gdrive_url_dir(gdrive, tmp_dir, dvc):
```

Comment:
Could you clarify why we need this test when we already have gdrive in `test_import_url_dir` above?

Reply:
I explained in the other conversion. I'll copy it here:

I've noticed, in the current tests, `GDriveDependency` is always getting a tree based on the remote tree, because the tests use `remote://` urls.

https://github.com/iterative/dvc/blob/7501302baa6f724a2e962446249044a563b44568/dvc/dependency/__init__.py#L65-L67

So, the case where `self._bucket` is `None` was never tested. This test using a `gdrive://` url solves this gap.

Reply:
@julsal After your recent change `self._bucket` shouldn't be used anywhere. If some path_info` is supplied, we should always use `path_info.bucket` and not use `self`. Which means that we can remove `self._bucket` shortcut and just not have to test it anymore.

Reply:
`self._bucket` was removed. The method `GDriveTree.gdrive_delete_file` receives `item_id` instead of `path_info`. Should I rely on `self.path_info.bucket` as I did, or should I change its signature to get `path_info` (public method)?

Should I remove the `test_import_gdrive_url_dir`? Removing `self._bucket` doesn't change the fact that `GDriveDependency` is always getting a tree based on the remote tree when we test with `remote://` urls. Additionally, it tests the particularities of `gdrive://ID`, as:

https://github.com/iterative/dvc/blob/e90ae72d41413ebc0ea8e96465d67073be113f42/dvc/tree/gdrive.py#L86-L87

if self._spath is empty, `is_relative_to` get lost:

https://github.com/iterative/dvc/blob/6324bd8a69ddb7ca4bd4d32f6e2ae2b368ce266d/dvc/tree/base.py#L330

This is a case I spotted because of the new test. The current test works because the url has `path`.